vault/external_tests/raft/raft_test.go

```diff
@@ -32,30 +33,50 @@ import (
 	"golang.org/x/net/http2"
 )
 
-func raftCluster(t testing.TB, joinNodes bool) *vault.TestCluster {
+type RaftClusterOpts struct {
+	DisableFollowerJoins bool
+	InmemCluster         bool
+	EnableAutopilot      bool
+}
+
+func raftCluster(t testing.TB, ropts *RaftClusterOpts) *vault.TestCluster {
+	if ropts == nil {
+		ropts = &RaftClusterOpts{}
+	}
+
 	conf := &vault.CoreConfig{
 		CredentialBackends: map[string]logical.Factory{
 			"userpass": credUserpass.Factory,
 		},
 	}
 
-	inmemCluster, err := vaultcluster.NewInmemLayerCluster("inmem-cluster", 3, hclog.New(&hclog.LoggerOptions{
-		Mutex: &sync.Mutex{},
-		Level: hclog.Trace,
-		Name:  "inmem-cluster",
-	}))
-	if err != nil {
-		t.Fatal(err)
+	var opts = vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
 	}
 
-	var opts = vault.TestClusterOptions{
-		HandlerFunc:   vaulthttp.Handler,
-		ClusterLayers: inmemCluster,
+	if ropts.InmemCluster {
+		inmemCluster, err := vaultcluster.NewInmemLayerCluster("inmem-cluster", 3, hclog.New(&hclog.LoggerOptions{
+			Mutex: &sync.Mutex{},
+			Level: hclog.Trace,
+			Name:  "inmem-cluster",
+		}))
+		if err != nil {
+			t.Fatal(err)
+		}
+		opts.ClusterLayers = inmemCluster
 	}
+
 	teststorage.RaftBackendSetup(conf, &opts)
-	if !joinNodes {
+	if ropts.DisableFollowerJoins {
 		opts.SetupFunc = nil
 	}
+
+	if !ropts.EnableAutopilot {
+		envVar := "VAULT_RAFT_AUTOPILOT_DISABLE"
```

Comment:
Could we use a config value instead of an ENV var for this? This will apply to all tests in the same test binary

Reply:
Despite being unset?

The main reason I was hesitant to add a config option is that it might be hard to deprecate it going forward, and we'll end up supporting this forever.

If we really want to go the config route, can we make it an undocumented thing, which we can backoff from (or make public) later?

Reply:
The tests seem to be doing the right thing with env var, but may be I am missing your point.

Reply:
The concern with the env var is that if you have two tests in the same package and test1 wants to disable autopilot, if the tests run in parallel then test2 might get it disabled too.

You avoid the concern about deprecation by keeping the env var for end users, and adding a config setting like Brian was saying - not something that goes in a config file, something that's only available to tests, e.g. a CoreConfig setting.

Reply:
Got it. Thanks for the suggestion Bk and for the explanation Nick. Done.